### ISSUE TITLE
Weirdness for setting name tags when importing collection with rule builder?

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#7611]
https://github.com/galaxyproject/galaxy/pull/6500/ indicates that this should be possible, but the options does not show up for me in a subset of cases and I'm wondering if this is an intentional decision, or something maybe overlooked.

Given the following URLs:

```
https://zenodo.org/record/1185122/files/GSM461176_untreat_single.counts
https://zenodo.org/record/1185122/files/GSM461177_untreat_paired.counts
https://zenodo.org/record/1185122/files/GSM461178_untreat_paired.counts
https://zenodo.org/record/1185122/files/GSM461179_treat_single.counts
https://zenodo.org/record/1185122/files/GSM461180_treat_paired.counts
https://zenodo.org/record/1185122/files/GSM461181_treat_paired.counts
https://zenodo.org/record/1185122/files/GSM461182_untreat_single.counts
```

- "Upload data: datasets", we can use the nametags.
- "Upload data: collections", nametags are unavailable and it isn't clear why.

We can build collections of similar shapes from datasets which have name tags, however we can't do this immediately in the rule builder.

Additionally name tags don't seem to be rendered for elements of collections?

![image](https://user-images.githubusercontent.com/458683/55076799-39fd7200-508e-11e9-96e3-4e7d5574ff1f.png)

And lastly, when the name tag option does appear, it doesn't behave like the other tagging options? We can't pull tag values from multiple columns? Maybe this is contrary to the intended use of "name" tags but, it fits with the use in practice sometimes? If people want to abuse name tags in different ways, it might be nice to permit it.

**Comment by mvdbeek** [comment]
Yeah, that would be good to allow uniformly, I think the reason for not having done this initially was that we didn't create a new HDA. Operating on the existing HDA would mean that you can accidentally tag "unrelated" datasets. But whenever we build a collection we should copy the HDA (I think we do that everywhere now), so we should allow setting tags as well (especially for group tags this would be very useful).

**Comment by hexylena** [comment]
> "Upload data: collections", nametags are unavailable and it isn't clear why.

Definitely still want this, I've got a history where I've done a dozen operations on two collections and cannot keep them straight, if I'd thought to tag the collections "#genes" or "#dna", it would be nicer now. (Yes I can do it manually but, I've already got a huge json blob for the uploader.)
